Update social media icons in email editor

NeosinneR opened the pull request:
## Description

1. Adds two new icons sets - official icons and their white version.
    <img width="319" alt="Screenshot 2025-06-03 at 21 59 33" src="https://github.com/user-attachments/assets/3bd50981-074e-48d4-bd16-635b44e1db00" />
2. Add new networks:
    - Behance
    - Bluesky
    - Discord
    - GitHub
    - Gravatar
    - Mastodon
    - Medium
    - Patreon
    - Reddit
    - RSS
    - Spotify
    - Telegram
    - Threads
    - TikTok
    - Tumblr
    - Twitch
    - Viemo
    - WhatsApp
    - WordPress
3. Update templates to use the new icon sets.
4. Hide all previous icon sets to discourage using them, but keep them for backward compatibility.

## Code review notes

_N/A_

## QA notes

It would be good to try all email templates to see if the social icon block looks good (e.g., there aren't white icons on light background).

## Linked PRs

_N/A_

## Linked tickets

STOMAIL-7323

## After-merge notes

_N/A_

## Tasks

- [x] I have added a changelog entry (pcNwfB-4Ov-p2#how-to-write-a-changelog)
- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] 🚫 I added sufficient test coverage
- [x] 🚫 I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:stomail-7323-update-social-media-icons-in-email-editor)

_The latest successful build from `stomail-7323-update-social-media-icons-in-email-editor` will be used. If none is available, the link won't work._

NeosinneR committed:
Add official social icons

STOMAIL-7323

NeosinneR committed:
Add white social icons

STOMAIL-7323

NeosinneR committed:
Simplify link label in social icons block

STOMAIL-7323

NeosinneR committed:
Add two new social icon sets

STOMAIL-7323

NeosinneR committed:
Sort social icons alphabetically

STOMAIL-7323

NeosinneR committed:
Allow selecting new social networks

STOMAIL-7323

NeosinneR committed:
Add fallback to missing icons for legacy icon sets

STOMAIL-7323

NeosinneR committed:
Hide legacy icons sets by default, with checkbox to show them

This is to keep backwards compatibility with existing emails but to direct new emails to use new icon sets

STOMAIL-7323

NeosinneR committed:
Update templates to use new social icons

STOMAIL-7323

NeosinneR committed:
Replace old Twitter with new X icon in templates

STOMAIL-7323

NeosinneR committed:
Add changelog

STOMAIL-7323

NeosinneR commented:
@lysyjan just a reminder about this PR :)

lysyjan reviewed (dismissed):
Very nice improvements.
I have tested the new icons, but I would appreciate further testing from QA.

Aschepikov commented:
Hi @NeosinneR, I see that there is a point to update the templates to use the new icon sets, but all templates currently use the legacy icon sets. Is this expected?
![image](https://github.com/user-attachments/assets/b4ff6877-1173-42c7-a3b1-9e3dbe652783)

NeosinneR commented:
@Aschepikov the new templates will only work for new installations. Existing websites will continue to have the old ones. 

This is because when the MailPoet is installed, templates are inserted from the code into the database. So when you update the plugin, any change in the templates is not seen, because we don't update the templates if they are already in the database (to prevent overriding any potential change that the users might have made). 

To test the templates, you'll need to install this build on a new website.

Aschepikov commented:
Great, running a new site

Aschepikov commented:
@NeosinneR on a new site there are two templates where the twitter/x icon is black on black
![image](https://github.com/user-attachments/assets/01dc7dc6-c562-44ac-9d0b-f2a51336f338)
![image](https://github.com/user-attachments/assets/8e411af0-8409-4e91-8345-c8c93d52b71e)
And on all templates the linkedin icon is not shown
![image](https://github.com/user-attachments/assets/40a15d4f-01f2-4b9c-8fac-f5d33a582a87)

NeosinneR committed:
Fix black X icon on dark background in Engineering and Birds templates

STOMAIL-7323

NeosinneR committed:
Fix incorrect path for Linkedin icon

STOMAIL-7323

NeosinneR commented:
@Aschepikov thanks for the review. I changed the two templates so they use white icons and fixed the LinkedIn icons (there was `I` instead of `i` in the URL).

Aschepikov commented:
@NeosinneR there are some conflicts can you repush it pls

Aschepikov reviewed (approved):
GJ!
